strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/cron/jobs/RegenerateMavenChecksumCronJobTestIT.java

```diff
@@ -49,161 +49,116 @@
                   ArtifactManagementTestExecutionListener.class })
     public void testRegenerateChecksumInRepository(@MavenRepository(repositoryId = REPOSITORY_SNAPSHOTS2)
-                                                   Repository repository,
+                                                           Repository repository,
                                                    @MavenTestArtifact(repositoryId = REPOSITORY_SNAPSHOTS2,
-                                                                      id = "org.carlspring.strongbox:strongbox-checksum-two",
-                                                                      versions = { "2.0-20190512.202015-1",
-                                                                                   "2.0-20190512.202101-2",
-                                                                                   "2.0-20190512.202203-3",
-                                                                                   "2.0-20190512.202311-4",
-                                                                                   "2.0-20190512.202601-5" })
-                                                  List<Path> artifactPaths)
+                                                           id = "org.carlspring.strongbox:strongbox-checksum-two",
+                                                           versions = { "2.0-20190512.202015-1",
+                                                                        "2.0-20190512.202101-2",
+                                                                        "2.0-20190512.202203-3",
+                                                                        "2.0-20190512.202311-4",
+                                                                        "2.0-20190512.202601-5" })
+                                                           List<Path> artifactPaths)
             throws Exception
     {
+        final String artifactBaseDir = "org/carlspring/strongbox/strongbox-checksum-two";
+
+        Map<String, String> cronJobConfigProperties = Maps.newHashMap();
+        cronJobConfigProperties.put("forceRegeneration", "false");
+
+        testRegenerateChecksum(repository,
+                               artifactPaths,
+                               artifactBaseDir,
+                               cronJobConfigProperties);
+    }
+
+    private void testRegenerateChecksum(Repository repository,
+                                        List<Path> artifactPaths,
+                                        String artifactBaseDir,
+                                        Map<String, String> cronJobConfigProperties)
+            throws Exception
+    {
+        final String storageId = repository.getStorage().getId();
+        final String repositoryId = repository.getId();
 
         final UUID jobKey = expectedJobKey;
         final String jobName = expectedJobName;
 
         RepositoryPath artifactPath = (RepositoryPath) artifactPaths.get(4);
```

Comment:
@ptirador why we use only fourth artifact from list? what are the rest for? I saw that you did this a while ago, but anyway, can you please clarify.

Reply:
oh, sorry seems this was made by @carlspring, [commit](https://github.com/strongbox/strongbox/commit/a4bd327222f804e2255b30747c099c050b1fc608)

Martin, why do we need 5 snapshots here if only one used?

Reply:
and it's seems not clear what is the difference between `testRegenerateChecksumInRepository` and `testRegenerateArtifactChecksum`

Reply:
You don't need all of them. Create and use just the ones you'll use.

Reply:
> and it's seems not clear what is the difference between testRegenerateChecksumInRepository and testRegenerateArtifactChecksum

I've created issue [#1442](https://github.com/strongbox/strongbox/issues/1442)